gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposeExtension.kt

```diff
@@ -41,5 +43,26 @@ abstract class ComposeExtension @Inject constructor(
      */
     val kotlinCompilerPluginArgs: ListProperty<String> = objects.listProperty(String::class.java)
 
+    /**
+     * A set of kotlin platform types to which the Compose Compiler plugin will be applied.
+     * By default, it contains all KotlinPlatformType values.
+     * It can be used to disable the Compose Compiler plugin for one or more targets:
+     * ```
+     * platformTypes.set(platformTypes.get() - KotlinPlatformType.native)
+     * ```
+     */
+    @Suppress("MemberVisibilityCanBePrivate")
+    val platformTypes: SetProperty<KotlinPlatformType> = objects.setProperty(KotlinPlatformType::class.java).also {
```

Comment:
make it private? WDYT?

Reply:
when it's public it can be conveniet to use it like this:

```
platformTypes.set(setOf(KotlinPlatformType.jvm))
```

to enable jvm only and disable for all other

Reply:
Better to provide only one API for setting this.

I would choose `platformTypes` over `excludePlatform`. Better to not have an exclude API as the only API (it isn't flexible/extendable). And if we have override-style API, it superseeds exclude-style API.

> platformTypes.set(platformTypes.get() - setOf(KotlinPlatformType.js, KotlinPlatformType.native))

> 2nd sample seems to be too verbose

In Gradle 8 (the current Gradle) users can write::

> platformTypes = platformTypes.get() - setOf(KotlinPlatformType.js, KotlinPlatformType.native)

Reply:
done